src/server/cards/adventures/Raze.js

```diff
@@ -0,0 +1,43 @@
+import Card from 'cards/Card';
+
+export default class Raze extends Card {
+  static cost = new Card.Cost({ coin: 2 });
+  static types = new Set(['Action']);
+  async onPlay(player) {
+    player.actions++;
+    const choice = await player.selectOption(['Trash Raze', 'Trash a card from your hand']);
+    let trashedCard;
+    switch (choice) {
+      case 0:
+        trashedCard = this;
+        await player.trash(this, player.playArea);
+        break;
+      case 1:
+        [trashedCard] = await player.selectCards({
+          min: 1,
+          max: 1,
+          message: 'Choose a card to trash',
+        });
+        await player.trash(trashedCard);
```

Comment:
trashedCard is never defined.

In case 1: put a check that they actually selected a card